config/main.py

```diff
@@ -474,9 +502,32 @@ def add_portchannel_member(ctx, portchannel_name, port_name):
 def del_portchannel_member(ctx, portchannel_name, port_name):
     """Remove member from portchannel"""
     db = ctx.obj['db']
+    port_channel = db.get_entry('PORTCHANNEL', portchannel_name)
+    if get_interface_naming_mode() == "alias":
+        port_name = interface_alias_to_name(port_name)
+        if port_name is None:
+            ctx.fail("'port_name' is None!")
+    
+    if len(port_channel) == 0:
+        ctx.fail("{} doesn't exists".format(portchannel_name))
+    
+    members = port_channel.get('members', [])
+    if port_name not in members:
+        if get_interface_naming_mode() == "alias":
+            port_name = interface_name_to_alias(port_name)
+            if port_name is None:
+                ctx.fail("'port_name' is None!")
+            ctx.fail("{} is not a member of {}".format(port_name, portchannel_name))
+        else:
+            ctx.fail("{} is not a member of {}".format(port_name, portchannel_name))
+    
+    members.remove(port_name)
+    if len(members) == 0:
+        del port_channel['members']
+    else:
+        port_channel['members'] = members
```

Comment:
same as above

Reply:
we need the `**members**` variable to delete the port channel when there are no members in it. AFAIK, we delete port channels when there are no members in it, not sure the SONiC use case. please advise. 

Thanks
Madhu

Reply:
addressed the review comments and updated the PR.  Thanks.

Reply:
there're alternative ways of checking if the port-channel has members or not. please do not use this approach. we are trying to make the code DRY by not having duplicated information on two places; this attribute won't be supported in the future release